src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -379,6 +379,7 @@ void ImageReplayer<I>::bootstrap() {
 
   {
     Mutex::Locker locker(m_lock);
+    request->get();
```

Comment:
Instead of inheriting from a RefCountedObj, you could use a `std::unique_ptr` to hold the pointer to the BootstrapRequest. The request object would be automatically deleted upon the destruction of the image replayer.